scriptworker/config.py

```diff
@@ -102,30 +109,35 @@ def check_config(config, path):
     return messages
 
 
-def create_config(path="config.json"):
+# create_config {{{1
+def create_config(config_path="scriptworker.json"):
     """Create a config from DEFAULT_CONFIG, arguments, and config file.
 
     Then validate it and freeze it.
 
     Args:
-        path (str, optional): the path to the config file.  Defaults to "config.json"
+        config_path (str, optional): the path to the config file.  Defaults to
+            "scriptworker.json"
 
     Returns:
-        tuple: (config dict, credentials dict)
+        tuple: (config frozendict, credentials dict)
```

Comment:
This is behavior I originally added to mozharness.  The idea is, `context` is used a lot, and `context.config` and `context.cot_config` as well.  If they're dicts, then they can be modified at runtime easily.  This may be intentional, if we want to change `context.config['foo'] = bar`.  Sometimes this is unintentional: we pass `context.config` to something by reference, and that function makes an unintentional change, and we end up with a modified `context.config` that causes weird behavior later on.

By changing `context.config` and `context.cot_config` to frozendicts with tuples instead of lists as child values, we force someone who wants to make a runtime change to explicitly cast to a dict before they can alter values; this is discouraged anyway, and we'll be more clear that that's what's happening.  Otherwise we're more certain that the runtime config will not change accidentally.